### ISSUE TITLE
[BeginHorizontal] and [ReorderableList] conflict

### ISSUE THREAD

**Issue by Vacui** [arimger/Unity-Editor-Toolbox#64]
```
[BeginHorizontal(labelToWidthRatio:0.1f)]
[SerializeField]
[ReorderableList(elementLabel:""), InLineEditor]
protected List<AIVehicleBehaviour> behaviours = new();

[EndHorizontal]
[SerializeField]
[ReorderableList(elementLabel:""), InLineEditor]
protected List<AIVehicleBehaviour> combatBehaviours = new();
```
![image](https://user-images.githubusercontent.com/45497567/209680752-557a76bd-e774-4025-ba79-94edffbe8eb7.png)
![image](https://user-images.githubusercontent.com/45497567/209681001-db1f6091-81f7-4bc8-b38f-d00e626713b9.png)

When increasing the number of elements on a **ReorderableList** it will reduce the size of another list if they are connected Horizontally.
I tested without all the fancy parameters, and I noticed that even with 2 simple **ReorderableList**, the one with more elements increases in width slightly the more you add to it. The combination of _labelToWidthRatio_ and _elementLabel_ probably exacerbates the issue.
The problem is not present when using 2 simple non-reorderable lists. But it has the problem that the right list overlaps slightly with the first in correspondence of the headers.
![image](https://user-images.githubusercontent.com/45497567/209681958-8defaf5a-414a-4a73-80b1-066a3e05508d.png)

From my tests, it seems the cause is the parameter _labelToWidthRatio_ in the **BeginHorizontal**. But removing it creates another problem of its own:
![image](https://user-images.githubusercontent.com/45497567/209681824-e82a752a-0c2f-4f06-8891-f9e42cc044b0.png)

That is what I found, I hope you can help. If you need more data just let me know.

Have a nice day and thanks for all your hard work.
Matteo

**Comment by arimger** [comment]
Hello,

thanks for the issue. I will try to investigate it and hopefully find a neat solution. Unfortunately, nesting horizontal and vertical groups is a bit tricky, so I can't promise anything.

**Comment by arimger** [comment]
Hello, I know it's late but I finally decided to change a bit how the horizontal layouts work so maybe this will be helpful for someone.

**What's the problem?** 

While creating a new horizontal layout I'm reusing Unity's `EditorGUILayout.BeginHorizontal/EndHorizontal` methods and I'm not analyzing what serialized properties are inside this layout. I don't want to do this to minimize the impact of the Toolbox package on performance. So basically, I don't know what should be the width of each element and Unity takes this responsibility while drawing/preparing layouts. Unfortunately, when calculating the needed width it also takes height into account that's why larger lists can take more space. 

**What's the solution?**

It's already implemented in the **develop** branch. I added a bunch of additional properties to manipulate horizontal groups:
- `bool ControlFieldWidth` - if false then nothing changes, and the layout will be completely auto-sized, if true we need to specify other properties.
- `int ElementsInLayout` - the most important one, using the available space I can calculate the best `EditorGUIUtility.fieldWidth` for this layout but I need to know how many elements are inside.
- `float WidthOffset/WidthOffsetPerElement` - additional offsets that can be used if you are using custom decorators within the layout, e.g. space.

That's my proposition on how to fix this issue, I will test it a bit and if everything is ok then will be merged together with the next release.
